### PR TITLE
fix: name of bamqc

### PR DESCRIPTION
**Pull request by christopher-schroeder**
### Description

The subcommand name was wrong in the description. Additionally add a region param fro target analyse.

### QC

* [x] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

### CODE REVIEW

**Comment by fgvieira** [comment]
Shouldn't target be an input file?
And can you also add a test using a target file?

**Comment by christopher-schroeder** [comment]
I thought about that. The problem is, that this parameter is optional. When developing a generic pipeline you would like to have this paramter for exomes and dont want to have it genomes. So you normally build a function. Something like

```
rule qualimap:
    input:
        bam="results/alignment/{sample}.bam",
        target=get_target
```

In the case of genomic data, get_target is None or "". But if defined as input, Snakemake will check the file for existence. But it cannot check for None or "". So I think the solution then would be

def qualimap_input(wc):
    ret = {"bam": "results/alignment/{wc.sample}.bam"}
    if target:
        ret["target"] = target

```
rule qualimap:
    input:
        unpack(qualimap_input)
```

I am not convinced about either of the 2 options and in the end I decided to go for the params method. But if you prefer the other method you can convince me easily.

**Comment by fgvieira** [comment]
I think the way to go would be option 2 and have it as an input.

File: bio/qualimap/bamqc/wrapper.py [hunk 1]
```diff
@@ -22,9 +22,13 @@
     del os.environ["DISPLAY"]
 
 extra = snakemake.params.get("extra", "")
+if target := snakemake.params.get("target"):
+    extra = f"{extra} --feature-file {target}"
```
Comment (johanneskoester) [inline bio/qualimap/bamqc/wrapper.py:26]: Can you please add this to the example Snakefile (can be commented out, but people have to see that it exists and they can use it)?

**Review by johanneskoester — CHANGES_REQUESTED** [review]

**Comment by johanneskoester** [comment]
> I thought about that. The problem is, that this parameter is optional. When developing a generic pipeline you would like to have this paramter for exomes and dont want to have it genomes. So you normally build a function. Something like
> 
> ```
> rule qualimap:
>     input:
>         bam="results/alignment/{sample}.bam",
>         target=get_target
> ```
> 
> In the case of genomic data, get_target is None or "". But if defined as input, Snakemake will check the file for existence. But it cannot check for None or "". So I think the solution then would be
> 
> ```
> def qualimap_input(wc):
>     ret = {"bam": "results/alignment/{wc.sample}.bam"}
>     if target:
>         ret["target"] = target
>     return ret
> 
> rule qualimap:
>     input:
>         unpack(qualimap_input)
> ```
> 
> I am not convinced of either of the 2 options and in the end I decided to go for the params method. But if you prefer the other method you can convince me easily.

It is also possible to return `[]` from the input function to indicate that the target is missing.

**Comment by fgvieira** [comment]
yes, it can also be option 1, if you return `None` when no target.
But it should be an input file

File: bio/qualimap/bamqc/wrapper.py [hunk 2]
```diff
@@ -22,9 +22,13 @@
     del os.environ["DISPLAY"]
 
 extra = snakemake.params.get("extra", "")
+if target := snakemake.params.get("target"):
```
Comment (johanneskoester) [inline bio/qualimap/bamqc/wrapper.py:25]: Can you please also add the new param to the example (e.g. commented out and mentioned as optional)?

**Review by johanneskoester — CHANGES_REQUESTED** [review]

**Review by fgvieira — APPROVED** [review]

**Pull request merged**